cetmix_tower_server/models/cx_tower_plan_line.py

```diff
@@ -187,3 +192,10 @@ def _skip(self, server, plan_log_record, **kwargs):
             is_skipped=True,
             **log_vals,
         )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            current_datetime = datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f")
```

Comment:
Please move this before the cycle starts

Reply:
you can use datetime from field:
```python
fields.Datetime.now()
```

Also use Server datetime format (I recommend using this method):
```python
from odoo.tools.misc import DEFAULT_SERVER_DATETIME_FORMAT
```

If this format does not suit you, then define your format as a constant in constants.py: 
```python
PLAN_LINE_DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S.%f"
```

Reply:
> Please move this before the cycle starts

I need unique data to create a reference (the "name" field is used to create a reference)

Reply:
> > Please move this before the cycle starts
> 
> I need unique data to create a reference (the "name" field is used to create a reference)

@dessanhemrayev  'name' field should'n be unique for an unique reference to be generated. This case is already handled in the [create](https://github.com/cetmix/cetmix-tower/blob/2711a9e613a0c3f7728cf176665e78f74ec8372a/cetmix_tower_server/models/cx_tower_reference_mixin.py#L137) method of the mixin.